### PR TITLE
Snow 998984 show help when no args

### PR DESCRIPTION
**Pull request by sfc-gh-mraba**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
MainCliSnowTyper callback was enriched witch ctx handling and discovery whether not invoked with subcommand.
SnowTyper has no_args_is_help set to True.

### CODE REVIEW

File: tests/__snapshots__/test_help_messages.ambr [hunk 1]
```diff
@@ -28,6 +59,35 @@
   ╰──────────────────────────────────────────────────────────────────────────────╯
   
   
+  '''
```
Comment (sfc-gh-turbaszek) [inline tests/__snapshots__/test_help_messages.ambr:62]: Let's add test for something else than top level, something  like `snow connection` just to be sure

Comment (sfc-gh-mraba) [reply tests/__snapshots__/test_help_messages.ambr:62]: I've added tests for all commands without subcomamnds.
These were skipped due to `list` `add` commands do not display help but execute action.

File: tests/test_help_messages.py [hunk 2]
```diff
@@ -4,20 +4,64 @@
...
+    yield no_command
+
     def _iter_through_commands(command, path):
         yield list(path)
         for subpath, subcommand in getattr(command, "commands", {}).items():
             path.append(subpath)
             yield from _iter_through_commands(subcommand, path)
             path.pop()
 
+    # yield []  # "snow" with no commands
```
Comment (sfc-gh-pczajka) [inline tests/test_help_messages.py:24]:
```suggestion

```

File: tests/test_help_messages.py [hunk 3]
```diff
@@ -4,20 +4,63 @@
...
+    ignore_plugins = ["render"]
+
+    no_command = []
+    yield no_command
+
+    def _iter_through_commands(command, path):
+        yield list(path)
+        if not plugin.plugin_name in ignore_plugins:
+            yield from _iter_through_commands(
+                spec.command, spec.full_command_path.path_segments
```
Comment (sfc-gh-pczajka) [inline tests/test_help_messages.py:61]: This falls into an infinite loop, did you mean `spec.full_command_path.path_segments + [subcommand]`?

File: tests/test_help_messages.py [hunk 4]
```diff
@@ -30,10 +73,10 @@ def _iter_through_commands(command, path):
 
 @pytest.mark.parametrize(
     "command",
-    iter_through_all_commands_paths(),
-    ids=(".".join(cmd) for cmd in iter_through_all_commands_paths()),
+    iter_through_all_commands_and_subcommands_paths(),
```
Comment (sfc-gh-pczajka) [inline tests/test_help_messages.py:76]: This should remain `iter_through_all_commands_paths`, otherwise both tests are essentially the same

**Review by sfc-gh-pjob — APPROVED** [review]

**Pull request merged**